lldb/source/Target/Process.cpp

```diff
@@ -1941,8 +1941,11 @@ size_t Process::ReadMemory(const AddressSpec &addr_spec, void *buf,
   }
   // We have an address that can't be resolved in the default address space, so
   // we need to call the overload that knows how to read from an address space.
-  llvm::Expected<AddressSpaceInfo> info = 
-      GetAddressSpaceInfo(addr_spec.GetSpaceName());
+  // Check if we have a numeric address space ID first.
+  llvm::Expected<AddressSpaceInfo> info =
+      addr_spec.GetSpaceId() ? GetAddressSpaceInfo(*addr_spec.GetSpaceId())
+                             : GetAddressSpaceInfo(addr_spec.GetSpaceName());
+
```

Comment:
We have a function in AddressSpec already for this. So we should use that:
```
  llvm::Expected<AddressSpaceInfo> AddressSpec::GetAddressSpaceInfo(lldb_private::Process &process) const;
```

Reply:
Done.